sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -194,7 +194,7 @@ case class ResolvedStar(expressions: Seq[NamedExpression]) extends Star {
   override def toString: String = expressions.mkString("ResolvedStar(", ", ", ")")
 }
 
-case class UnresolvedGetField(child: Expression, fieldName: String) extends UnaryExpression {
+case class UnresolvedGetField(child: Expression, fieldExpr: Expression) extends UnaryExpression {
```

Comment:
can you add some javadoc for this class. at the very least, we should say this can be used to get a field out of struct/map/array.